### DISCUSSION TITLE
How to customize window class names on the Windows platform?

### DISCUSSION THREAD

**Discussion by xushushun** [AvaloniaUI/Avalonia#15545] [General]
My software runs on the Windows platform and needs to integrate with third-party programs that require access to my window titles and class names. The class name of Avalonia windows is randomly generated, such as "Avalonia-5413e27f-445a-4065-bee6-4aa9bb8dd950", and I cannot customize it. Is there any solution? I found that the code for this feature is located in src/Windows/Avalonia.Win32/WindowImpl.cs, but packaging and building NuGet packages is very challenging for me.

**Comment by stevemonaco** [comment]
`lpszClassName` must be unique within a process. There's no access to this as Avalonia necessarily abstracts many platform-specific details and uses the GUID as a way to uniquely ID the Window within the process without having external knowledge. The Window also needs unregistered via `UnregisterClass`, so you probably shouldn't try to change it with your own win32api calls.

The third-party software is a bit unfortunate in how it picks windows if the class name is required as a sole strategy. It should support other strategies if it's a general purpose app. If it requires specific support, then it should use IPC.

**Comment by xushushun** [comment]
Thank you for your response. I understand now that I shouldn't dwell on this.
